### DISCUSSION TITLE
Client Side Actions Looping

### DISCUSSION THREAD

**Discussion by poirazis** [Budibase/budibase#7779] [Ideas]
Scenario : 
I have an awesome screen where the user does awesome stuff.
I also have a button.
When this button is pressed, i want to iterate the data provider, and execute the equivalent number of "Save Row" to insert multiple rows in a table.

Button OnClick
- Loop -> Source : Data Provider Rows ( Or Table.SelectedRows )
- Save Row -> (Create new row in table)
- Loop End
- Refresh Data Provider(s)

Is there a way ( other than via an Automation and the array passed as param ) to do this client side ?
Is it even a good idea / thing to have ?

**Comment by melohagan** [comment]
I imagine this would be quite useful! I've raised a feature request here: https://github.com/Budibase/budibase/issues/7785
